### ISSUE TITLE
Incorrect loading of entity with a list attribute when entity cache is enabled

### ISSUE THREAD

**Issue by pasha9819** [cuba-platform/cuba#3187]
### Environment
- Platform version: 7.2.10

#### Minimal reproducible example:
- Create empty project wirh two entities: Tree and Leaf
```java
@Table(name = "LISTANDCACHE_TREE")
@Entity(name = "listandcache_Tree")
@NamePattern("%s|name")
public class Tree extends StandardEntity {

    @OnDeleteInverse(DeletePolicy.UNLINK)
    @OnDelete(DeletePolicy.UNLINK)
    @OneToMany(mappedBy = "tree")
    @Composition
    private List<Leaf> leafs;

    @Column(name = "NAME")
    private String name;
	
    // get, set
}

@Table(name = "LISTANDCACHE_LEAF")
@Entity(name = "listandcache_Leaf")
@NamePattern("%s|description")
public class Leaf extends StandardEntity {

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "TREE_ID")
    private Tree tree;

    @Column(name = "DESCRIPTION")
    private String description;
	
    // get, set
	
}
```
- enable entity cache
```
eclipselink.cache.shared.listandcache_Tree = true
eclipselink.cache.shared.listandcache_Leaf = true
```
- create Tree entity with one composite Leaf entity
- save Tree entity
- try to open Tree Editor screen

#### Expected behavior
- Tree Editor screen will be open  
#### Actual behavior
- `IllegalStateException: Changed instance com.company.listandcache.entity.Tree-cac725da-31da-9edc-fdd1-d9a560393623 [managed,detached] in read-only transaction`

This exception occurs in `com.haulmont.cuba.core.sys.persistence.PersistenceImplSupport$ContainerResourceSynchronization.beforeCommit(PersistenceImplSupport.java:423) ~[cuba-core-7.2.10.jar:7.2.10]` occurs due to changes in the `leafs` attribute of the Tree entity, but there were no explicit changes to this attribute after saving to the database

**Comment by pasha9819** [comment]
https://user-images.githubusercontent.com/29690565/117969894-0e56a800-b339-11eb-9754-fc791fa108f9.mp4

**Comment by plakhov** [comment]
Fixed in #3166
